lib/isomorphic-http-client.js

```diff
@@ -1,188 +1,232 @@
 const fetch = require('cross-fetch');
 const CryptoJS = require('crypto-js');
 const qs = require('qs');
-const _ = require("lodash");
-
-const applicationJsonTypes = ['application/vnd.api+json', 'application/json', 'application/problem+json'];
-const hasContentType = function(contentType, types){
+const _ = require('lodash');
+
+const applicationJsonTypes = [
+  'application/vnd.api+json',
+  'application/json',
+  'application/problem+json',
+];
+const hasContentType = function (contentType, types) {
   if (!contentType || !types) return null;
-  return _.reduce(types, function(acc, val) {
-    return acc || contentType.indexOf(val) !== -1;
-  }, false);
+  return _.reduce(
+    types,
+    function (acc, val) {
+      return acc || contentType.indexOf(val) !== -1;
+    },
+    false,
+  );
 };
 
-exports.get = function(params){
+exports.get = function (params) {
   return exports._send(exports._makeBaseRequest(params, 'GET'));
 };
 
-exports.post = function(params){
+exports.post = function (params) {
   return exports._send(exports._makeBaseRequest(params, 'POST'));
 };
 
-exports.delete = function(params){
+exports.delete = function (params) {
   return exports._send(exports._makeBaseRequest(params, 'DELETE'));
 };
 
-exports.put = function(params){
+exports.put = function (params) {
   return exports._send(exports._makeBaseRequest(params, 'PUT'));
 };
 
-exports.patch = function(params){
+exports.patch = function (params) {
   return exports._send(exports._makeBaseRequest(params, 'PATCH'));
 };
 
-function shouldOverwriteSignature(newSignature, headers){
+function shouldOverwriteSignature (newSignature, headers) {
   return newSignature && !/^Bearer/.test(headers.Authorization || '');
 }
 
-exports._makeBaseRequest = function(params, method){
+function shouldSetUserAgent (headers) {
+  return _.isNil(headers['user-agent']) && typeof window === undefined;
+}
+
+exports._makeBaseRequest = function (params, method) {
   let oauthHeader = null;
   params.method = method;
   if (params.oauth) {
-    oauthHeader = exports._createOAuthHeader(params, params.oauth.consumer_key, params.oauth.consumer_secret);
+    oauthHeader = exports._createOAuthHeader(
+      params,
+      params.oauth.consumer_key,
+      params.oauth.consumer_secret,
+    );
   }
   const headers = params.headers || {};
-  if (shouldOverwriteSignature(oauthHeader, headers)) headers.Authorization = oauthHeader;
+  if (shouldOverwriteSignature(oauthHeader, headers))
+    headers.Authorization = oauthHeader;
+  if (shouldSetUserAgent(headers))
+    headers['user-agent'] = ['FortuneClient', process.env.npm_package_version]
```

Comment:
Does this one refer to fortune-client actually? must be a single global var shared across all dependencies..

Reply:
oh yeah that makes sense lol